src/server/active_tran_server.cpp

```diff
@@ -281,25 +281,43 @@ active_tran_server::is_page_server_connected () const
 }
 
 void
-active_tran_server::init_log_page_broker ()
+active_tran_server::init_page_brokers ()
 {
-  m_log_page_broker.reset (new cublog::page_broker ());
+  m_log_page_broker.reset (new page_broker<log_page_type> ());
+  m_data_page_broker.reset (new page_broker<data_page_type> ());
 }
 
 void
-active_tran_server::finalize_log_page_broker ()
+active_tran_server::finalize_page_brokers ()
 {
   m_log_page_broker.reset ();
+  m_data_page_broker.reset ();
 }
 
-cublog::page_broker &
-active_tran_server::get_log_page_broker ()
+template<>
+page_broker<log_page_type> &
+active_tran_server::get_page_broker ()
 {
-  assert (m_log_page_broker);
+  if (!m_log_page_broker)
+    {
+      m_log_page_broker.reset (new page_broker<log_page_type> ());
+    }
```

Comment:
The page brokers should be already initialized before they're first needed. Best to use a safe-guard and to avoid hiding an initialization error.

```suggestion
  assert (m_log_page_broker != nullptr);
```